editing_and_imputation/editing/totals_components/technical_specification.md

```diff
@@ -0,0 +1,316 @@
+an error detection method.
+
+### 5.4 Determine Error Detection Method (Stage 5)
+
+Stage 5 determines whether the detected difference in the
+provided components and totals falls into the absolute or percentage
+difference thresholds specified and therefore needs to be
+automatically correct or the values need to be manually corrected.
+
+When only Absolute Difference Threshold is set to 0 or above
+the threshold is checked against the absolute value calculated
+in step 4. We complete this action in section 5.4.1.
+
+When only Percentage Difference Threshold is set to 0 or above
+the threshold is checked against the predictive total. The percentage
+threshold is the sum of the received components plus and minus the
+Percentage Difference Threshold specified. If this threshold is not
+satisfied then we require manual editing and the method stops.
+
+### 5.4.1 Check Absolute Difference Threshold (Stage 5a)
+
+The absolute difference between the target total and
+the components must be less than or equal to the absolute difference threshold.
+
+When the Absolute Difference Threshold check indicates the
+difference needs to be automatically corrected the method
+continues to stage 6 in section 5.5.
+
+When the Absolute Difference Threshold check indicates the
+correction needs to be manually applied the method will check
+if a Percentage Difference Threshold needs to be checked.
+
+Note: Before we leave this stage we need to check
+that zero error condition 3 is satisfied i.e. If
+target total = 0 and components sum > 0 and amend total = TRUE:
+The total should be corrected if the difference observed is within
+the tolerances determined by the detection method (see section 5.5).
+Else, the difference should be flagged for manual checking.
+
+### 5.4.2 Check Percentage Difference Threshold (Stage 5b)
+
+If the total for the predictive is within the low and high
+percentage then we go to stage 6 in section 5.5.
+Otherwise, we require manual editing and stop the method.
+
+### 5.5 Error Correction (Stage 6)
+
+This section covers the error correction aspect of the process.
+When the method has reached this part of the processing earlier checks
+have determined that either the total or the components must
+be automatically corrected.
+
+When the input parameter amend_total indicates that the total
+must be amended we automatically correct the total.
+
+Where the amend_total indicates the components need to corrected,
+we use the total value and the precision for the component
+automatic correction calculations.
+
+Expanding on this if we correct the total then we set the final
+total in the output data equivalent to the total, and the
+final values for all the components match their originals. We would now
+return a totals corrected marker.
+
+However, if the components are corrected to match the received
+total based on the weighting of the original input component values,
+then we return a components corrected marker.
+
```

Comment:
Can you mention the algorithm used to correct the components using the total_value, sum_of_components terminology